source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -103,10 +103,10 @@ void CheckRequestUtils::setRequestTime(envoy::service::auth::v3::AttributeContex
 }
 
 void CheckRequestUtils::setHttpRequest(
-    envoy::service::auth::v3::AttributeContext::HttpRequest& httpreq, uint64_t stream_id,
+    envoy::service::auth::v3::AttributeContext::HttpRequest& httpreq,
     const StreamInfo::StreamInfo& stream_info, const Buffer::Instance* decoding_buffer,
     const Envoy::Http::RequestHeaderMap& headers, uint64_t max_request_bytes, bool pack_as_bytes) {
-  httpreq.set_id(std::to_string(stream_id));
+  httpreq.set_id(getHeaderStr(headers.RequestId()));
```

Comment:
This header may be disabled through the [generate_request_id](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto.html?highlight=generate_request_id) config option. What would happen in that case?